core/src/main/scala/kafka/admin/TopicCommand.scala

```diff
@@ -120,7 +120,10 @@ object TopicCommand extends Logging {
       opts.reportUnavailablePartitions && hasUnavailablePartitions(partitionDescription)
     }
     private def hasUnderMinIsrPartitions(partitionDescription: PartitionDescription) = {
-      partitionDescription.isr.size < partitionDescription.minIsrCount
+      if (partitionDescription.leader.isDefined)
```

Comment:
TopicCommandWithAdminClientTest.testDescribeUnderMinIsrPartitionsMixed will fail. If leader is not available, the partition must be a under-ISR one.

Reply:
Hmm, good point. Since `min.insync.replicas` cannot be 0, I guess we could simplify this.
```scala
partitionDescription.leader.isEmpty || partitionDescription.isr.size < partitionDescription.minIsrCount
```